src/lib/catchy.ts

```diff
@@ -0,0 +1,24 @@
+// this utility function wraps a promise and catches errors of a specific type.
+// it makes your code more catch-strict and less error - prone
+// since the `try-catch` must parse the errors to be caught
+async function catchy<T, E extends new (message?: string) => Error>(
+  promise: Promise<T>,
+  errsToCatch?: Array<E>,
+): Promise<[undefined, T] | [InstanceType<E>]> {
+  try {
+    const v = await promise;
+    return [undefined, v] as [undefined, T];
+  } catch (err) {
+    if (errsToCatch === undefined) {
+      return [err as InstanceType<E>];
+    }
+
+    if (errsToCatch.some((e) => err instanceof e)) {
+      return [err as InstanceType<E>];
+    }
+
```

Comment:
Could you please add some console.error here to indicate that the "catchy" function is not working properly? 

I just want to make sure that future developer who will use this function don't need to debug their code when using this function but its not working as intended 

Because if the catchy is working properly, it shouldn't reach this line, right?

Reply:
nice catch, agree with this statement, i'll add the custom params for enable debugging to reduce the unwanted output, wdyt?

Reply:
![image](https://github.com/user-attachments/assets/b12acc1b-fccb-4647-b928-b156d4e8df3b)

I've been added the verbose mode with a guard clause to make sure the developer aware of the error log

Reply:
Ohh i think better use `logLevel` like the suggestion from the bot

Reply:
More levels don't mean to be used by checking all of them, since the verbose only logs the error, I think it's enough for now, wdyt?